tiledb/sm/crypto/test/unit_crypto.cc

```diff
@@ -493,50 +517,113 @@ TEST_CASE("Crypto: Test AES-256-GCM", "[crypto][aes]") {
   }
 }
 
-TEST_CASE("Crypto: Test MD5", "[crypto][md5]") {
-  SECTION("- Basic") {
-    std::string expected_checksum = "e99a18c428cb38d5f260853678922e03";
-    std::string text_to_checksum = "abc123";
-    ConstBuffer input_buffer(
-        text_to_checksum.data(), text_to_checksum.length());
-    Buffer output_buffer;
-    CHECK(output_buffer.realloc(Crypto::MD5_DIGEST_BYTES).ok());
-    CHECK(Crypto::md5(&input_buffer, &output_buffer).ok());
-
-    unsigned char* digest =
-        reinterpret_cast<unsigned char*>(output_buffer.data());
-    char md5string[33];
-    for (uint64_t i = 0; i < output_buffer.alloced_size(); ++i) {
-      sprintf(&md5string[i * 2], "%02x", (unsigned int)digest[i]);
-    }
-    CHECK(
-        memcmp(
-            expected_checksum.data(), md5string, expected_checksum.length()) ==
-        0);
+static std::vector<uint8_t> from_hex(const std::string& str) {
+  std::vector<uint8_t> vec;
+  for (size_t i = 0; i < str.length(); i += 2) {
+    char byte_str[3] = {str[i], str[i + 1], '\0'};
+    vec.push_back((uint8_t)std::strtoul(byte_str, nullptr, 16));
   }
+  return vec;
 }
 
-TEST_CASE("Crypto: Test SHA256", "[crypto][sha256]") {
-  SECTION("- Basic") {
-    std::string expected_checksum =
-        "6ca13d52ca70c883e0f0bb101e425a89e8624de51db2d2392593af6a84118090";
-    std::string text_to_checksum = "abc123";
-    ConstBuffer input_buffer(
-        text_to_checksum.data(), text_to_checksum.length());
-    Buffer output_buffer;
-    CHECK(output_buffer.realloc(Crypto::SHA256_DIGEST_BYTES).ok());
-    CHECK(Crypto::sha256(&input_buffer, &output_buffer).ok());
-
-    unsigned char* digest =
-        reinterpret_cast<unsigned char*>(output_buffer.data());
-    char shastring[65];
-    for (uint64_t i = 0; i < output_buffer.alloced_size(); ++i) {
-      sprintf(&shastring[i * 2], "%02x", (unsigned int)digest[i]);
-    }
+static std::string to_hex(const uint8_t* data, uint64_t length) {
+  std::stringstream ss;
+  for (uint64_t i = 0; i < length; i++) {
+    ss << std::hex << std::setw(2) << std::setfill('0') << (int)data[i];
+  }
+  return ss.str();
+}
+
+// Test that the given input has the expected hash value in
+// hex. The function is generic over the hash and the input type.
+template <class Hash>
+static void test_hash(
```

Comment:
An unqualified name "test" does not state what's being tested. Test names should describe what they are rather than how they work. In this case there's not a lot of difference.

In this case the test is validating an argument/return-value test vector.

Reply:
Renamed.